Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs

```diff
@@ -38,51 +39,47 @@ public LawDisplay(EntityUid uid, SiliconLaw law, HashSet<string>? radioChannels)
         if (!_entityManager.TryGetComponent<SpeechComponent>(uid, out var speech) || speech.SpeechSounds is null)
             return;
 
-        var localButton = new Button
+        // New unified button to work with drop down list
+        var stateLawButton = new Button
         {
-            Text = Loc.GetString("hud-chatbox-select-channel-Local"),
-            Modulate = Color.DarkGray,
+            Text = Loc.GetString("hud-borgui-state-law"),
+            Modulate = Color.LightBlue,
             StyleClasses = { "chatSelectorOptionButton" },
             MinHeight = 35,
-            MinWidth = 75,
+            MinWidth = 100
         };
 
-        localButton.OnPressed += _ =>
+        stateLawButton.OnPressed += _ =>
         {
-            _chatManager.SendMessage($"{lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Local);
-        };
-
-        LawAnnouncementButtons.AddChild(localButton);
-
-        if (radioChannels == null)
-            return;
-
-        foreach (var radioChannel in radioChannels)
-        {
-            if (!_prototypeManager.TryIndex<RadioChannelPrototype>(radioChannel, out var radioChannelProto))
-                continue;
-
-            var radioChannelButton = new Button
+            _sawmill.Debug($"Selected channel detected as {selectedChannel}");
+            // Local and Binary channels are added to the list outside of the Radio component, so we handle them uniquely
+            if (selectedChannel == "Local")
+                _chatManager.SendMessage($"{lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Local);
+            else if (selectedChannel == "Binary")
+                _chatManager.SendMessage($"{SharedChatSystem.RadioChannelPrefix}b {lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Radio);
+            else
             {
-                Text = Loc.GetString(radioChannelProto.Name),
-                Modulate = radioChannelProto.Color,
-                StyleClasses = { "chatSelectorOptionButton" },
-                MinHeight = 35,
-                MinWidth = 75,
-            };
+                if (radioChannels == null)
+                    return;
 
-            radioChannelButton.OnPressed += _ =>
-            {
-                switch (radioChannel)
+                foreach (var radioChannel in radioChannels)
                 {
-                    case SharedChatSystem.CommonChannel:
-                        _chatManager.SendMessage($"{SharedChatSystem.RadioCommonPrefix} {lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Radio); break;
-                    default:
-                        _chatManager.SendMessage($"{SharedChatSystem.RadioChannelPrefix}{radioChannelProto.KeyCode} {lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Radio); break;
+                    if (radioChannel == selectedChannel)
```

Comment:
Why are you doing a foreach loop but then filtering all but one element? This can be much simpler.